packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1077,257 +1077,275 @@ void ArbCore::operator()() {
+                    break;
                 }
+            }
 
-                // Exit now that profiling is complete
-                break;
+            if (core_machine->status() == MachineThread::MACHINE_ABORTED) {
+                // Just reset status so machine can be restarted
+                core_machine->clearError();
             }
-        }
 
-        if (core_machine->status() == MachineThread::MACHINE_ABORTED) {
-            // Just reset status so machine can be restarted
-            core_machine->clearError();
-        }
+            if (core_machine->status() == MachineThread::MACHINE_NONE) {
+                // Start execution of machine if new message available
+                auto success = runMachineWithMessages(
+                    execConfig, coreConfig.message_process_count, true);
+                if (!success) {
+                    break;
+                }
+            }
 
-        if (core_machine->status() == MachineThread::MACHINE_NONE) {
-            // Start execution of machine if new message available
-            auto success = runMachineWithMessages(
-                execConfig, coreConfig.message_process_count, true);
-            if (!success) {
-                break;
+            for (size_t i = 0; i < logs_cursors.size(); i++) {
+                if (logs_cursors[i].status == DataCursor::REQUESTED) {
+                    ReadTransaction tx(data_storage);
+                    handleLogsCursorRequested(tx, i, cache);
+                }
             }
-        }
 
-        for (size_t i = 0; i < logs_cursors.size(); i++) {
-            if (logs_cursors[i].status == DataCursor::REQUESTED) {
-                ReadTransaction tx(data_storage);
-                handleLogsCursorRequested(tx, i, cache);
+            if (save_checkpoint) {
+                ReadWriteTransaction tx(data_storage);
+                save_checkpoint_status = saveCheckpoint(tx);
+                tx.commit();
+                save_checkpoint = false;
             }
-        }
 
-        if (save_checkpoint) {
-            ReadWriteTransaction tx(data_storage);
-            save_checkpoint_status = saveCheckpoint(tx);
-            tx.commit();
-            save_checkpoint = false;
+            if (!machineIdle() || message_data_status != MESSAGES_READY) {
+                // Machine is already running or no new messages, so sleep for a
+                // short while
+                std::this_thread::sleep_for(std::chrono::milliseconds(
+                    coreConfig.idle_sleep_milliseconds));
+            }
         }
 
-        if (!machineIdle() || message_data_status != MESSAGES_READY) {
-            // Machine is already running or no new messages, so sleep for a
-            // short while
-            std::this_thread::sleep_for(
-                std::chrono::milliseconds(coreConfig.idle_sleep_milliseconds));
-        }
+        std::cerr << "Exiting main ArbCore thread" << std::endl;
```

Comment:
Let's move this down to under the try-catch so we can search logs for it consistently

Reply:
:+1: